File: app/src/main/java/pemwa/com/javadevsnairobi/util/NetworkUtility.java
package pemwa.com.javadevsnairobi.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtility {

    public static boolean isConnected(Context context) {

        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        boolean connected = false;

        if (connectivityManager != null) {
            NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
            connected = (networkInfo != null) && (networkInfo.isConnectedOrConnecting());
        }

        return connected;
    }
}

File: app/src/main/java/pemwa/com/javadevsnairobi/util/NetworkUtility.kt
package pemwa.com.javadevsnairobi.util

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

object NetworkUtility {

    fun isConnected(context: Context): Boolean {

        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        var connected = false

        if (connectivityManager != null) {
            val networkInfo = connectivityManager.activeNetworkInfo
            connected = networkInfo != null && networkInfo.isConnectedOrConnecting
        }

        return connected
    }
}
